src/form_beam.cu

```diff
@@ -255,19 +256,25 @@ __global__ void vmBeamform_kernel( cuDoubleComplex *Jv_Q,
     int ant  = threadIdx.x; /* The (ant)enna number */
     int nant = blockDim.x;  /* The (n)_umber of (ant)ennas */
 
-    /*// GPU profiling
-    clock_t start, stop;
-    double setup_t, detect_t, sum_t, stokes_t;
-    if ((p == 0) && (ant == 0) && (c == 0) && (s == 0)) start = clock();*/
-
     // Organise dynamically allocated shared arrays (see tag 11NSTATION for kernel call)
     extern __shared__ double arrays[];
-
-    cuDoubleComplex *ex  = (cuDoubleComplex *)(&arrays[1*nant]);
-    cuDoubleComplex *ey  = (cuDoubleComplex *)(&arrays[3*nant]);
-    cuDoubleComplex *Nxx = (cuDoubleComplex *)(&arrays[5*nant]);
-    cuDoubleComplex *Nxy = (cuDoubleComplex *)(&arrays[7*nant]);
-    cuDoubleComplex *Nyy = (cuDoubleComplex *)(&arrays[9*nant]);
+    // These SHOULD be *aligned* on integer numbers of 4-byte blocks.
+
+    // NOTE: Because these are complex-doubles, each takes up 2*sizeof(double), hence the stride of 2 here.
+    /* Given that we need to ensure access alignment on the 4-byte boundaries (CUDA requirement), we have
+       to make sure that the array access corresponds to an integer number of sizeof(double), which
+       means we need to use even indexes to access memory. This ensures for all `nant` that we access
+       within the memory boundaries. This StackOverflow post helped us figure this out:
+         
+            https://stackoverflow.com/questions/70765553/cuda-shared-memory-alignement-in-documentation
+
+       (Previously, the indexes were: 1*nant, 3*nant, etc.)
+     */
+    cuDoubleComplex *ex  = (cuDoubleComplex *)(&arrays[0*nant]);
+    cuDoubleComplex *ey  = (cuDoubleComplex *)(&arrays[2*nant]);
+    cuDoubleComplex *Nxx = (cuDoubleComplex *)(&arrays[4*nant]);
+    cuDoubleComplex *Nxy = (cuDoubleComplex *)(&arrays[6*nant]);
+    cuDoubleComplex *Nyy = (cuDoubleComplex *)(&arrays[8*nant]);
```

Comment:
just want to be sure, this is the only actual change right? Others seems to just be debugging messages and comments

Reply:
Yes, that's correct.